### PR TITLE
Isolate mapping computation to simplify presentation

### LINKED ISSUE #101: NIZK security properties
The rationale for the NIZK proof could be made more clear. We should add this somewhere to the protocol overview, with details in the security considerations. We should also extract the "mapping" computation to a separate section so it can be reviewed independently of the rest of the protocol details.

### PR DESCRIPTION
**Pull request by chris-wood**
The only remaining piece of "crypto" in the issuance flow description is the blind RSA bits. I'm not yet sure what we want to do with that, but I feel like this sort of extraction is useful for clarity. I'm happy to hear alternative proposals for presentations, too!

Closes #101.

cc @FredericJacobs

### CODE REVIEW

File: draft-private-access-tokens.md [hunk 1]
```diff
@@ -948,26 +932,12 @@ was invalid in the process.
...
-index = ORIGIN_SECRET * mapping_input
-mapping_index = SerializeElement(index)
-~~~
-
-If DeserializeElement fails, or if AccessTokenRequest.mapping_key is the identity element, the Issuer
-rejects the request with a 400 error.
+If the Issuer is willing to give a token to the Client, the Issuer decrypts
+AccessTokenRequest.encrypted_origin_name to discover "origin". If this fails, the Issuer rejects
+the request with a 400 error. Otherwise, the Issuer validates and processes the token request
+with ORIGIN_SECRET correspoding to the designated Origin as described in {{issuer-stable-mapping}}.
```
Comment (dvorak42) [inline draft-private-access-tokens.md:938]: nit: corresponding

File: draft-private-access-tokens.md [hunk 2]
```diff
@@ -1084,11 +1049,173 @@ enc, context = SetupBaseR(enc, skI, "AccessTokenRequest")
...
+  ---------------------->
+                             (request, proof)
+                         ---------------------->
+                                (response)
+                         <----------------------
+~~~
+
+The protocol for computing this functionality is divided into sections for
+each of the participants. {{client-stable-mapping}} describes Client behavior
+for initating the computation with its per-Client secret, {{issuer-stable-mapping}}
```
Comment (dvorak42) [inline draft-private-access-tokens.md:1078]: nit: initiating

File: draft-private-access-tokens.md [hunk 3]
```diff
@@ -1084,11 +1049,173 @@ enc, context = SetupBaseR(enc, skI, "AccessTokenRequest")
...
+## Stable Mapping Computation {#stable-mapping}
+
+This section describes the Client, Mediator, and Issuer behavior in computing
+the stable mapping based on client identity and origin name. At a high level,
+this functionality computes y = F(x, k), where x is a per-Client secret and
+k is a per-Origin secret, subject to the following constraints:
+
+- The Mediator only learns y if the Client in possession of x engages with the protocol;
+- The Client cannot engage in the protocol for private input x' that is not equal to x; and
+- The Issuer does not learn x, nor does it learn when two requests correspond to the same private value x.
```
Comment (dvorak42) [inline draft-private-access-tokens.md:1061]: The wording here is a little confusing, maybe reword cannot engage with mediator prevents? Also say something about what people learn about k?

File: draft-private-access-tokens.md [hunk 4]
```diff
@@ -1084,11 +1049,173 @@ enc, context = SetupBaseR(enc, skI, "AccessTokenRequest")
...
+    raise InvalidParameterError
+
+// Verify the proof
+proof = DeserializeProof(client_origin_index_proof)
+valid = SchnorrVerify(blind_generator, blind_target, proof)
+if not valid:
+   raise InvalidProofError
+~~~
+
+Given an Issuer response client_origin_index_result, Client blind, and Client public
```
Comment (dvorak42) [inline draft-private-access-tokens.md:1158]: Maybe subsection each half of the mediator protocol? Wonder if it should just be split in protocol order:

Client Behavior
Mediator Behavior (Client Key Verification)
Issuer Behavior
Mediator Behavior (Output)

Though this seems fine.

**Review by dvorak42 — APPROVED** [review]
Looks reasonable.

**Comment by chris-wood** [comment]
Thanks @dvorak42! I applied your suggestions.

**Review by tfpauly — APPROVED** [review]
